source/material_model/melt_boukare.cc

```diff
@@ -0,0 +1,1260 @@
+                                                                                     "Third coefficients for specific heat polynomial");
+
+          //Check all lists have the correct length
+          AssertThrow(endmember_names.size() == endmember_states.size(),
+                      ExcMessage("One of the lists that define the endmember parameters does not have the "
+                                 "correct size.  Please check your parameter file."));
+
+
+          // All of these are molar fractions
+          // Check that all compositional fields we need exist
+          AssertThrow(this->introspection().compositional_name_exists("molar_Fe_in_solid"),
+                      ExcMessage("Material model melt boukare only works if there is a "
+                                 "compositional field called 'molar_Fe_in_solid'."));
+
+          if (this->include_melt_transport())
+            {
+              AssertThrow(this->introspection().compositional_name_exists("porosity"),
+                          ExcMessage("Material model melt boukare only works if there is a "
+                                     "compositional field called porosity."));
+              AssertThrow(this->introspection().compositional_name_exists("molar_Fe_in_melt"),
+                          ExcMessage("Material model melt boukare only works if there is a "
+                                     "compositional field called 'molar_Fe_in_melt'."));
+            }
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    MeltBoukare<dim>::create_additional_named_outputs (MaterialModel::MaterialModelOutputs<dim> &out) const
+    {
+      if (this->get_parameters().use_operator_splitting
+          && out.template get_additional_output<ReactionRateOutputs<dim>>() == nullptr)
+        {
+          const unsigned int n_points = out.viscosities.size();
+          out.additional_outputs.push_back(
+            std::make_unique<MaterialModel::ReactionRateOutputs<dim>> (n_points, this->n_compositional_fields()));
+        }
+
+      if (out.template get_additional_output<BoukareOutputs<dim>>() == nullptr)
+        {
+          const unsigned int n_points = out.viscosities.size();
+          out.additional_outputs.push_back(
+            std::make_unique<MaterialModel::BoukareOutputs<dim>> (n_points));
+        }
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    ASPECT_REGISTER_MATERIAL_MODEL(MeltBoukare,
+                                   "melt boukare",
+                                   "A material model that implements a melting model following Boukare"
+                                   "et al and uses it to compute the "
+                                   "material parameters required for the modelling of melt transport, "
+                                   "including a source term for the porosity according to a simplified.")
```

Comment:
simplified ... . Also maybe include a note to look up the details in our paper. Just take the text from the header file.